File: frontend/src/app/models/order.ts
import {Client} from "./client";
import {FoodPlace} from "./food-place";
import {OrderStatus} from "./enums/order-status";

export class Order {
  constructor(public orderId?: number,
              public totalPrice?: number,
              public dishes?: { [key: string]: number },
              public client?: Client,
              public place?: FoodPlace,
              public orderDate?: string,
              public orderStatus?: OrderStatus | string) {
  }
}

File: src/main/java/com/project/waiterless/models/Order.java
package com.project.waiterless.models;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDate;
import java.util.Map;

@Builder
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "orders")
public class Order {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int orderId;
    int totalPrice;
    @ElementCollection
    @Column(name = "quantity", nullable = false)
    @MapKeyJoinColumn
    Map<String, Integer> dishes;
    @ManyToOne
    Client client;
    @ManyToOne
    FoodPlace place;
    LocalDate orderDate;
    String orderStatus;

    @Override
    public String toString() {
        return "Order{" +
                "orderId=" + orderId +
                ", totalPrice=" + totalPrice +
                ", dishes=" + dishes +
                ", orderDate=" + orderDate +
                ", orderStatus='" + orderStatus + '\'' +
                '}';
    }
}
